deployment.yml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: DemoK8
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: DemoK8
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+  minReadySeconds: 5
+  template:
+    metadata:
+      labels:
+        app: DemoK8
+    spec:
+      containers:
+      - name: DemoK8
+        image: gcr.io/${{ secrets.GKE_PROJECT }}/harsittheking001/docker-spring-k8-demo:1.0-SNAPSHOT
```

Comment:
as we are not fetching the image from docker hub, right now gcr is our docker registry, lets keep it at 
`gcr.io/PROJECT_ID/IMAGE:TAG`